src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs

```diff
@@ -368,6 +369,29 @@ public async Task<IActionResult> Delete(string typeParameter, string idParameter
             return FhirResult.NoContent().SetETagHeader(response.WeakETag);
         }
 
+        /// <summary>
+        /// Deletes the specified resource
+        /// </summary>
+        /// <param name="typeParameter">The type.</param>
+        /// <param name="hardDelete">A flag indicating whether to hard-delete the resource or not.</param>
+        /// <param name="maxDeleteCount">Specifies the maximum number of resources that can be deleted.</param>
+        [HttpDelete]
+        [Route(KnownRoutes.ResourceType)]
+        [AuditEventType(AuditEventSubType.Delete)]
+        public async Task<IActionResult> ConditionalDelete(string typeParameter, [FromQuery] bool hardDelete, [FromQuery(Name = KnownQueryParameterNames.Count)] int? maxDeleteCount)
+        {
+            IReadOnlyList<Tuple<string, string>> conditionalParameters = GetQueriesForSearch();
+
+            DeleteResourceResponse response = await _mediator.Send(new ConditionalDeleteResourceRequest(typeParameter, conditionalParameters, hardDelete, maxDeleteCount.GetValueOrDefault(1)), HttpContext.RequestAborted);
```

Comment:
>maxDeleteCount.GetValueOrDefault(1) [](http://example.com/codeflow?start=154&length=35)

Maybe it's worth to allow pass null here?

let's say I pass nothing, and I have two matching items, I would get Exception as result.

let's say I pass 2 and I have four items, now I don't get exception and 2 out 4 got deleted.

Now I pass 1 and I still get back to exception.

Feels weird...

Reply:
`_count` is not technically used on delete requests. I was talking to @CaitlinV39 about this too. I think we should default to a `Single row` deletion behavior unless you specify its ok to delete more. I think we can just document what should happen